File: Project 7 Main/Project7_CryptoBot_Dev/services/quant/src/crypto_quant_ai/monitoring/otel.py

from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import SimpleSpanProcessor
from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter

def init_tracer(service_name: str, otlp_endpoint: str = "http://otel-collector:4318/v1/traces"):
    provider = TracerProvider()
    trace.set_tracer_provider(provider)
    processor = SimpleSpanProcessor(OTLPSpanExporter(endpoint=otlp_endpoint))
    provider.add_span_processor(processor)
    return trace.get_tracer(service_name)

File: legacy_original_project7_main/Project7_CryptoBot_Dev/apps/backend/src/otel.ts

import { NodeSDK } from '@opentelemetry/sdk-node';
import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';

export function startOtel() {
  const url = process.env.OTLP_URL || 'http://otel-collector:4318/v1/traces';
  const exporter = new OTLPTraceExporter({ url });
  const sdk = new NodeSDK({ traceExporter: exporter, instrumentations: [getNodeAutoInstrumentations()] });
  sdk.start();
}
